flux-typeck/src/type_env.rs

```diff
@@ -125,8 +125,8 @@ impl TypeEnv {
             LookupResult::Ref(RefKind::Mut, _) => {
                 panic!("cannot move out of `{place:?}`, which is behind a `&` reference")
             }
-            LookupResult::Ref(RefKind::Shr, _) => {
-                panic!("cannot move out of `{place:?}`, which is behind a `&mut` reference")
+            LookupResult::Ref(RefKind::Shr, _) | LookupResult::Box(_) => {
```

Comment:
It's ok to move out of `Box`, e.g., 


```rust
fn foo(x: Box<Vec<i32>>) -> Vec<i32> {
    *x
}
```

Reply:
hmm -- ok so what should this code be? `move_place` ?

Reply:
Changing the code to the following seems to work?

```rust
    pub fn move_place(&mut self, rcx: &mut RefineCtxt, gen: &mut ConstrGen, place: &Place) -> Ty {
        match self.bindings.lookup_place(rcx, gen, place) {
            LookupResult::Ptr(path, ty) => {
                self.bindings.update(&path, Ty::uninit());
                ty
            }
            LookupResult::Ref(RefKind::Mut, _) => {
                panic!("cannot move out of `{place:?}`, which is behind a `&mut` reference")
            }
            LookupResult::Ref(RefKind::Shr, _) => {
                panic!("cannot move out of `{place:?}`, which is behind a `&` reference")
            }
            LookupResult::Box(ty) => ty,
        }
    }
```

Reply:
(also added a test for this in `box01.rs`)

Reply:
added comment!